packages/neo-one-client-core/src/user/DapiUserAccountProvider.ts

```diff
@@ -0,0 +1,427 @@
+    hash: Hash256String,
+    options: TransactionOptions = {},
+    sourceMaps: Promise<SourceMaps> = Promise.resolve({}),
+  ): Promise<TransactionResult<RawInvokeReceipt, InvocationTransaction>> {
+    return this.invokeCompleteSendBase(
+      contract,
+      method,
+      params,
+      paramsZipped,
+      hash,
+      options,
+      sourceMaps,
+      this.provider,
+    );
+  }
+
+  public async invokeRefundAssets(
+    contract: AddressString,
+    method: string,
+    params: ReadonlyArray<ScriptBuilderParam | undefined>,
+    paramsZipped: ReadonlyArray<readonly [string, Param | undefined]>,
+    hash: Hash256String,
+    options: TransactionOptions = {},
+    sourceMaps: Promise<SourceMaps> = Promise.resolve({}),
+  ): Promise<TransactionResult<RawInvokeReceipt, InvocationTransaction>> {
+    return this.invokeRefundAssetsBase(
+      contract,
+      method,
+      params,
+      paramsZipped,
+      hash,
+      options,
+      sourceMaps,
+      this.provider,
+    );
+  }
+
+  public async invokeClaim(
+    _contract: AddressString,
+    _method: string,
+    _params: ReadonlyArray<ScriptBuilderParam | undefined>,
+    _paramsZipped: ReadonlyArray<readonly [string, Param | undefined]>,
+    _options?: TransactionOptions,
+    _sourceMaps?: Promise<SourceMaps>,
+  ): Promise<TransactionResult<TransactionReceipt, ClaimTransaction>> {
+    throw new NotImplementedError('invokeClaim');
+  }
+
+  public async call(
+    network: NetworkType,
+    contract: AddressString,
+    method: string,
+    params: ReadonlyArray<ScriptBuilderParam | undefined>,
+  ): Promise<RawCallReceipt> {
+    await this.initPromise;
+
+    return this.provider.call(network, contract, method, params);
+  }
+
+  protected async invokeRaw<T extends TransactionReceipt>({
+    options = {},
+    invokeMethodOptions,
+    verify = true,
+    transfers = [],
+    rawInputs = [],
+    rawOutputs = [],
+    scripts,
+    onConfirm,
+    reorderOutputs = (outs) => outs,
+  }: InvokeRawOptions<T>): Promise<TransactionResult<T, InvocationTransaction>> {
+    await this.initPromise;
+    const { from, networkFee, systemFee } = this.getTransactionOptions(options);
+    if (invokeMethodOptions === undefined) {
+      throw new InvalidArgumentError('InvokeMethodOptions', 'invokeMethodOptions', invokeMethodOptions);
+    }
+    const { contract, invokeMethod: method, params } = invokeMethodOptions;
+
+    const { gas } = await this.checkSystemFees({
+      script: clientUtils.getInvokeMethodScript({
+        address: contract,
+        method,
+        params,
+      }),
+      from,
+      networkFee,
+      systemFee,
+      provider: this.provider,
+      scripts,
+      rawInputs,
+      rawOutputs,
+    });
+
+    const { inputs, outputs } = await this.getTransfersInputOutputs({
+      transfers,
+      from,
+      gas: networkFee.plus(gas),
+      provider: this.provider,
+    });
+
+    const { txid } = await this.dapi.invoke({
```

Comment:
It seems like the only concrete differences between this method and the equivalent in `LocalUserAccountProvider` are line 309 and 338-350 vs. the `sendTransaction` bit in `LocalUserAccountProvider`.